ivy/functional/ivy/linear_algebra.py

```diff
@@ -380,26 +383,173 @@ def diagonal(
         last two dimensions and appending a dimension equal to the size of the resulting
         diagonals. The returned array must have the same data type as ``x``.
 
-    Examples
-    --------
+    Functional Examples
+    ------------------
+
+    With :code:`ivy.Array` inputs:
+    
     >>> x = ivy.array([[1., 2.],\
                        [3., 4.]])
     >>> d = ivy.diagonal(x)
     >>> print(d)
     ivy.array([1., 4.])
+
+    >>> x = ivy.array([[1., 2.],\
+                       [3., 4.]])
     >>> d = ivy.diagonal(x, 1)
     >>> print(d)
     ivy.array([1.])
 
-    A 3-D Example
     >>> x = ivy.array([[[1., 2.],\
                         [3., 4.]],\
                        [[5., 6.],\
                         [7., 8.]]])
-    >>> d = ivy.diagonal(x, 0, 0, 1)
+    >>> d = ivy.diagonal(x)
     >>> print(d)
-    ivy.array([[1., 7.],
-               [2., 8.]])
+    ivy.array([[1., 4.],
+               [5., 8.]])
+
+    >>> y = ivy.arange(9).reshape((3, 3))
```

Comment:
Thanks for all the changes!

I was doing one look-through and I noticed that for some of your examples - particularly the ones that start with `ivy.arange().reshape()` - you haven't actually used an `ivy.array` input or `ivy.native_array` input in these.